Misc gulp tweaks

erikkemperman opened the pull request:
I found several little issues and thought I'd combine them in a single PR. If you like I can squash these into a single commit, but review is probably easier this way.

erikkemperman commented:
I added a commit to force a rebuild -- [this one](https://travis-ci.org/jhipster/generator-jhipster/jobs/106528543) failed for reasons I suspect are out of my hands:
```sh
The command "$JHIPSTER_INSTALL/01-install-docker-compose.sh" failed and exited with 18 during .

Your build has been stopped
```

Is the travis infrastructure being modified?

pascalgrimaud commented:
Sometimes, Travis is tired :construction_worker: 
You can close/reopen your PR or wait one of us to restart the only build which failed

erikkemperman commented:
Good to know, thanks!

Actually I think the last commit might be helpful, I imagine the `.tmp` folder of an existing project might get quite large, and should not be needlessly packaged into future wars...

Of course this only makes sense if the first commit were adopted.

erikkemperman commented:
I am running into some more little miscellaneous things, would it be preferred I add onto this PR, or is it better if I create separate ones?

gmarziou commented:
I think it's OK to add them to this PR as this is part our migration to gulp only.

erikkemperman commented:
All right, will do that, thanks. I am aware that I'm kind of spamming you guys with PRs, for a newcomer... But I'm setting up a project with jhipster and some of the things I "fix" seem like they might benefit others. Just let me know if it's too much too soon, or something :-)

deepu105 commented:
Sharing is always good so keep them coming :)
On 3 Feb 2016 05:37, "Erik Kemperman" <notifications@github.com> wrote:

> All right, will do that, thanks. I am aware that I'm kind of spamming you
> guys with PRs, for a newcomer... But I'm setting up a project with jhipster
> and some of the things I "fix" seem like they might benefit others. Just
> let me know if it's too much too soon, or something :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2766#issuecomment-178837379>
> .
>

deepu105 commented on app/index.js:
```diff
@@ -1025,11 +1025,9 @@ module.exports = JhipsterGenerator.extend({
+            } else {
+                this.template(WEBAPP_DIR + 'content/css/main.css', WEBAPP_DIR + 'content/css/main.css');
             }
-            // this css file will be overwritten by the sass generated css if sass is enabled
```
Why did you change this? This was on purpose this way

erikkemperman replied:
I made sure that sass, if enabled, is always run after `yo jhipster`, so this placeholder is no longer needed.

deepu105 replied:
we also had similar code before but there were lot of tickets regarding run failure and we wanted something to be there to startup even if build fails for any reason. I have had weird issue of build not running after project generation many times. SO its better to have this by default can you plz put it back

deepu105 commented on app/cleanup.js:
```diff
@@ -10,5 +10,6 @@ module.exports = {
 function cleanupOldFiles(generator, javaDir, testDir, resourceDir, webappDir, testJsDir) {
     if (generator.isJhipsterVersionLessThan('2.27.1')) {
         //removeFile and removeFolder methods should be called here for files and folders to cleanup
+        generator.removefolder('.tmp');
```
Not required coz next release will be major so we dont expect people to use on existing projects

erikkemperman replied:
Ah, good point, I will remove this commit!

deepu105 commented on app/index.js:
```diff
@@ -1452,16 +1450,24 @@ module.exports = JhipsterGenerator.extend({
-                    '\n ...and generate the Angular constants with:' +
-                    '\n' + chalk.yellow.bold('gulp ngconstant:dev')
+                    '\nOr do all of the above:' +
+                    '\n ' + chalk.yellow.bold('gulp install') +
```
Nice :)

deepu105 commented:
Let us know once the PR is ready for merge, i reviewed current diff looks good except for the 2 comments

deepu105 commented:
@erikkemperman the `gulp wiredep` task doesnt seem to inject any new bower component items into index.html can you chek plz? cc @ngasull

deepu105 commented:
@erikkemperman @ngasull oops my bad gulp wiredep works correctly

deepu105 commented:
Also we need to exclude the js files from bootstrap-sass folder during wiredep

erikkemperman commented:
> Also we need to exclude the js files from bootstrap-sass folder during wiredep

Not sure what you mean... Wiredep only inserts `bower_components/bootstrap-sass/assets/javascripts/bootstrap.js` which we need, right?

erikkemperman commented:
@deepu105 All right I've removed the commit that removed `.tmp` on re-generate, and just rebased everything onto latest `master`. 

Let me know if there's anything else you'd like to be changed (I think I've replied to all your comments). Otherwise I'm going to let this PR as it is, have to switch to another project today.

deepu105 commented:
its injecting 2 bootstrap files, one from normal bootstrap and one form sass, its causing issues

deepu105 commented:
i need to do some quick refactor so ill merge this and correct it

erikkemperman commented:
That's odd, for me it only injects a single script, from `bootstrap-sass`. From `app/templates/_bower.json`

```js
<%_ if (useSass) { _%>
        "bootstrap-sass": "3.3.5",
        <%_ } else { _%>
        "bootstrap": "3.3.5",
        <%_ } _%>
```
So I'm not seeing this problem on a newly created project...

erikkemperman commented:
Thanks for merging! I noticed you didn't wait for the travis tests, some of which are failing intermittently it seems...

deepu105 commented:
ya
